json-endpoints/class.wpcom-json-api-post-endpoint.php

```diff
@@ -65,35 +65,6 @@ function __construct( $args ) {
 		parent::__construct( $args );
 	}
 
-	function is_metadata_public( $key ) {
-		if ( empty( $key ) )
-			return false;
-
-		// Default whitelisted meta keys.
-		$whitelisted_meta = array( '_thumbnail_id' );
-
-		/**
-		 * Filters the meta keys accessible by the REST API.
-		 * @see https://developer.wordpress.com/2013/04/26/custom-post-type-and-metadata-support-in-the-rest-api/
-		 *
-		 * @module json-api
-		 *
-		 * @since 2.2.3
-		 *
-		 * @param array $whitelisted_meta Array of metadata that is accessible by the REST API.
-		 */
- 		if ( in_array( $key, apply_filters( 'rest_api_allowed_public_metadata', $whitelisted_meta ) ) )
```

Comment:
We use that filter in a few places in Jetpack. If it's not used anymore, we should remove it from there, and add the filter to the list of deprecated hooks in `class.jetpack.php`.